### PR TITLE
Edit ActivityEditCommand in the Developer Guide

### PR DESCRIPTION
**Pull request by matheril**
Updates the steps involved in this command to reflect the fact that it now supports optional arguments. Also added a sequence diagram.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -498,6 +498,45 @@ The general workflow of the `activity /delete` command is as follows:
...
+5. The `Profile#getSession` method is called to retrieve the `Session` object which the activity that the user wishes to edit is stored in.
+    - If the session does not exist, a message indicating that there is no such session is printed using `TextUi#printlnMessage` and control is given back to `SplitLah`.
+    - Else, the `Session` object that the activity is stored in is returned.
+6. `Session#getActivity` is called on the `activityId` provided by the user to fetch the `Activity` object representing
+   the activity to be edited. 
+    - If the activity does not exist, a message indicating that there is no such activity is printed using `TextUi#printlnMessage` and control is given back to `SplitLah`.
+    - Else, the `Activity` object to be edited is returned and stored temporarily as `oldActivity`.
+7. `ActivityEditCommand#retrieveDetailsFromOldActivity` is called on `oldActivity` to retrieve existing activity details
+   for delimiters not supplied by the user. This allows ActivityEditCommand to reuse existing details for details the user
```
Comment (froststein) [inline docs/DeveloperGuide.md:527]: Perhaps put ActivityEditCommand in code blocks

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -498,6 +498,45 @@ The general workflow of the `activity /delete` command is as follows:
...
+   does not wish to edit.
+8. `ActivityEditCommand#updateCostAndCostList` and `ActivityEditCommand#validateCostListAndInvolvedList` are called to 
+   create a list of costs and validate them. These costs are necessary to recreate the activity after deleting it.
+9. `ActivityEditCommand#addAllActivityCost` is called to add the respective costs to each `Person` object involved in the activity.
+     - A dummy `activityId` that cannot be used by any other activity is when adding these costs.
+10. `Session#removeActivity()` method is invoked to remove `oldActivity` from the list of activities stored.
+11. A new `Activity` object is created using the new activity details and stored as `newActivity`.
+12. `Session#addActivity` is called to add `newActivity` to the session.
+13. `ActivityEditCommand#updateDummyActivityIdsInActivityCosts` is called on the session to update all dummy activityIds added by `ActivityEditCommand#addAllActivityCost` to
```
Comment (froststein) [inline docs/DeveloperGuide.md:536]: Perhaps use ...all dummy activity unique identifiers ... instead?

**Review by froststein — CHANGES_REQUESTED** [review]
Generally looks good, just need to have some minor changes

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (matheril) [reply docs/DeveloperGuide.md:527]: Resolved

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (matheril) [reply docs/DeveloperGuide.md:536]: Resolved. Also changed line 532: A dummy activity unique identifier that cannot be used... to reflect this.

**Review by warrencxw — APPROVED** [review]
Looks good to merge

**Review by froststein — APPROVED** [review]
Looks good to merge

**Pull request merged**